mcs/class/System/Microsoft.CSharp/CSharpCodeGenerator.cs

```diff
@@ -119,14 +117,12 @@ private CompilerResults FromFileBatch (CompilerParameters options, string[] file
 			}
 
 			try {
-				mcs.BeginOutputReadLine ();
 				mcs.BeginErrorReadLine ();
 				mcs.WaitForExit();
 				
 				results.NativeCompilerReturnValue = mcs.ExitCode;
 			} finally {
-				mcs.CancelErrorRead ();
-				mcs.CancelOutputRead ();
+				stderr_completed.WaitOne (TimeSpan.FromSeconds (30));
```

Comment:
maybe make this 15 seconds instead? seems like a long enough time.

Reply:
I'm erring on the side of safety here. In practice, it rarely ever takes more than 500ms. And it's better to be safe than sorry should we hit a situation when the delay is longer.